paramak/parametric_components/blanket_poloidal_segment.py

```diff
@@ -0,0 +1,312 @@
+    """
+
+    def __init__(
+        self,
+        segments_angles=None,
+        num_segments=7,
+        length_limits=None,
+        nb_segments_limits=None,
+        segments_gap=0.0,
+        **kwargs
+    ):
+        super().__init__(
+            **kwargs
+        )
+        self.num_segments = num_segments
+        self.length_limits = length_limits
+        self.nb_segments_limits = nb_segments_limits
+        self.segments_angles = segments_angles
+        self.segments_gap = segments_gap
+        self.segments_cutters = None
+
+    @property
+    def segments_angles(self):
+        return self._segments_angles
+
+    @segments_angles.setter
+    def segments_angles(self, value):
+        if value is not None:
+            if self.start_angle is not None or self.stop_angle is not None:
+                msg = "start_angle and stop_angle attributes will be " + \
+                    "ignored if segments_angles is not None"
+                warnings.warn(msg, UserWarning)
+            elif self.num_segments is not None:
+                msg = "num_segment attribute will be ignored if " + \
+                    "segments_angles is not None"
+                warnings.warn(msg, UserWarning)
+        self._segments_angles = value
+
+    @property
+    def num_segments(self):
+        return self._num_segments
+
+    @num_segments.setter
+    def num_segments(self, value):
+        if value is not None:
+            self.num_points = value + 1
+        self._num_segments = value
+
+    @property
+    def segments_cutters(self):
+        self.create_segment_cutters()
+        return self._segments_cutters
+
+    @segments_cutters.setter
+    def segments_cutters(self, value):
+        self._segments_cutters = value
+
+    def get_angles(self):
+        """Get the poloidal angles of the segments.
+
+        Returns:
+            list: the angles
+        """
+        if (self.length_limits, self.nb_segments_limits) != (None, None):
+            angles = segments_optimiser(
+                self.length_limits, self.nb_segments_limits,
+                self.distribution, (self.start_angle, self.stop_angle),
+                stop_on_success=True
+            )
+        elif self.segments_angles is None:
+            angles = np.linspace(
+                self.start_angle, self.stop_angle,
+                num=self.num_segments + 1)
+        else:
+            angles = self.segments_angles
+        return angles
+
+    def find_points(self):
+        points = super().find_points(angles=self.get_angles())
+
+        # every points straight connections
+        for p in points:
+            p[-1] = 'straight'
+        self.points = points[:-1]
+
+    def create_solid(self):
+        solid = super().create_solid()
+        segments_cutters = self.segments_cutters
+        if segments_cutters is not None:
+            solid = cut_solid(solid, segments_cutters)
+        self.solid = solid
+        return solid
+
+    def create_segment_cutters(self):
```

Comment:
Do you think we could pull this shape out and have it as another parametric shape. It will be usefull when making the upper and lower caps of the first wall.

The caps are showing fig 1, stage 7 (right hand side) https://iopscience.iop.org/article/10.1088/1741-4326/ab0016/pdf

Reply:
I see.  The thing is these cutters have a safety factor so their lengths don't match the thickness of the blanket.

What could be done though is computing the difference: blanket_with_no_gaps - blanket_with_gaps = top_and_lower_caps.

Reply:
For the blanketcutterstar we just set the width to a default number which was high
https://paramak.readthedocs.io/en/latest/paramak.parametric_components.html#blanketcutterstar

Reply:
Yep, here it's tailored to the local thickness to be generic.
What do you think about making the caps from the intersection between blanket_with_no_gaps and cutters ?

Reply:
I think the caps need a small gap, so perhaps a cut between the blanket with a small gap and a blanket with a larger gap. A similar thing is done here to mage the firstwall https://github.com/ukaea/paramak/blob/0554e6f4f2d84b6e688bcaa19d3ee056acd6948b/paramak/parametric_reactors/segmented_blanket_ball_reactor.py#L126-L184